c/src/lib/libbsp/arm/shared/stm32fxxxx/spi/hal-spi.c

```diff
@@ -0,0 +1,507 @@
+  HAL_GPIO_Init(pin_config[spi_instance].port, &GPIO_InitStruct);
+}
+
+
+/**
+ * SPI Initalize Instance
+ */
+static int stm32_spi_init(stm32_spi_bus * bus)
+{
+   rtems_status_code sc;
+
+   _Assert(bus != NULL);
+
+   /* Initialize GPIO pins for SPI */
+   stm32_spi_gpio_init(bus->instance);
+
+/*********************** Initialize SPI ***************************/
+   /* Set the SPI parameters */
+   bus->handle->Instance               = stm32_spi_get_spi_instance(bus->instance);
+
+   bus->handle->Init.BaudRatePrescaler = SPI_BAUDRATEPRESCALER_4;
+   bus->handle->Init.Direction         = SPI_DIRECTION_1LINE;
+   bus->handle->Init.CLKPhase          = SPI_PHASE_1EDGE;
+   bus->handle->Init.CLKPolarity       = SPI_POLARITY_LOW;
+   bus->handle->Init.CRCCalculation    = SPI_CRCCALCULATION_DISABLE;
+   bus->handle->Init.CRCPolynomial     = 7;
+   bus->handle->Init.DataSize          = SPI_DATASIZE_16BIT;
+   bus->handle->Init.FirstBit          = SPI_FIRSTBIT_MSB;
+   bus->handle->Init.NSS               = SPI_NSS_SOFT;
+   bus->handle->Init.TIMode            = SPI_TIMODE_DISABLE;
+
+   bus->handle->Init.Mode = SPI_MODE_MASTER;
+   if(HAL_SPI_Init(&bus->handle) != HAL_OK){
+
+      (*bus->base.destroy)(&bus->base);
+      rtems_set_errno_and_return_minus_one(EIO);
+   }
+/******************************************************************/
+
+/********************* Install SPI vectors ***********************/
+   sc = rtems_interrupt_handler_install(
+         stm32_get_spi_vec_num(bus->instance),
+         "SPI Event Insturrpt",
+         RTEMS_INTERRUPT_UNIQUE,
+         stm32_spi_event_irq,
+         &bus->handle
+   );
+
+   if (sc != RTEMS_SUCCESSFUL) {
+
+      (*bus->base.destroy)(&bus->base);
+      rtems_set_errno_and_return_minus_one(EIO);
+   }
+/******************************************************************/
+   return 0;
+}
+
+/**
+ * spi Bus Destroy and De-initialise
+  */
+static int stm32_spi_deinit_destroy(stm32_spi_bus * bus)
+{
+     rtems_status_code sc;
+
+     _Assert(bus != NULL);
+
+     sc = rtems_interrupt_handler_remove(stm32_get_spi_vec_num(bus->instance), stm32_spi_event_irq, &bus->handle);
+     _Assert(sc == RTEMS_SUCCESSFUL);
+     (void) sc;
+
+     spi_bus_destroy_and_free(&bus->base);
+   return (HAL_SPI_DeInit(&bus->handle));
+}
+
+
+/* Register SPI Driver to RTEMS */
+int stm32_bsp_register_spi(void)
+{
+   rtems_status_code sc;
+   int err,inst_num=0;
+   char bus_path[12];
+
+   for(inst_num=0; inst_num < MAX_SPI_INSTANCES; inst_num++)
+   {
+      if(true == SPI_Enable[inst_num]) // Check if to enable this instance or not
+      {
+         bus[inst_num] = (stm32_spi_bus *) spi_bus_alloc_and_init(sizeof(*bus[inst_num]));
+
+         if (bus[inst_num] == NULL) {
+            return -ENOMEM;
+         }
+         bus[inst_num]->base.init       = stm32_spi_init;
+         bus[inst_num]->base.transfer   = stm32_spi_transfer;
+         bus[inst_num]->base.destroy      = stm32_spi_deinit_destroy;
```

Comment:
line them up or dont try to line them up. I personally don't care just maintain consistency